pyro/poutine/infer_config_poutine.py

```diff
@@ -0,0 +1,52 @@
+from __future__ import absolute_import, division, print_function
+
+from .poutine import Messenger, Poutine
+
+
+class InferConfigMessenger(Messenger):
+    """
+    Modifies contents of the infer kwarg at sample sites
+    """
+    def __init__(self, config_fn=None):
```

Comment:
I think there should be no default value for `config_fn`. If we absolutely need a default value, it should be `lambda site: {}`.